src/carts/src/main/java/mushop/carts/CartRepositoryDatabaseImpl.java

```diff
@@ -0,0 +1,145 @@
+    
+    /** The name of the backing collection */
+    private final String collectionName;    
+    
+    /** Pool of reusable database connections */
+    protected PoolDataSource pool;
+    
+    /** Used to automatically convert a Cart object to and from JSON */
+    private Jsonb jsonb;
+    
+    public CartRepositoryDatabaseImpl(Config config) {
+        try {
+            System.setProperty("oracle.jdbc.fanEnabled", "false");
+            String dbName = config.get("OADB_SERVICE").asString().get();
+            String url = "jdbc:oracle:thin:@" + dbName + "?TNS_ADMIN=${TNS_ADMIN}";
+            pool = PoolDataSourceFactory.getPoolDataSource();
+            pool.setMaxStatements(50);
+            pool.setConnectionFactoryClassName("oracle.jdbc.pool.OracleDataSource");
+            pool.setURL(url);
+            pool.setUser(config.get("OADB_USER").asString().get());
+            pool.setPassword(config.get("OADB_PW").asString().get());
+            collectionName = config.get("OADB_CARTS_COLLECTION").asString().get();
+
+            // Create the carts collection if it does not exist
+            try (OracleConnection con = (OracleConnection) pool.getConnection()) {
+                OracleDatabase db = SODA.getDatabase(con);
+                OracleCollection col = db.openCollection(collectionName);
+                if (col == null) {
+                    // Create a collection where the document id is not generated automatically.
+                    // It is OK if multiple processes try to create the collection at the
+                    // same time. The collection will simply be returned by createCollection() if it
+                    // already exists.
+                    OracleDocument collMeta = SODA.createMetadataBuilder().keyColumnAssignmentMethod("client").build();
+                    col = db.admin().createCollection(collectionName, collMeta);
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        
+        jsonb = JsonbBuilder.create();
+    }
+
+    @Override
+    public boolean deleteCart(String id) {
+        try (Connection con = pool.getConnection()) {
+            OracleDatabase db = SODA.getDatabase(con);
+            OracleCollection col = db.openCollection(collectionName);
+            int ct = col.find().key(id).remove();
+            return ct > 0;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Cart getById(String id) {
+        try (Connection con = pool.getConnection()) {
+            OracleDatabase db = SODA.getDatabase(con);
+            OracleCollection col = db.openCollection(collectionName);
+            OracleDocument doc = col.findOne(id);
+            return doc == null ? null : jsonb.fromJson(doc.getContentAsString(), Cart.class);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
If there's no connectivity to the DB, should not throw an exception. Should report to the health check endpoint the DB status .

Reply:
Yes, this is an issue.  As discussed on Slack, will look into this and fix in a follow-up merge.